### PR TITLE
Bundle kubemark in test tarball instead of server tarball

### LINKED ISSUE #28635: Do we need to ship the kubemark binary?
No end users will actually use that.  This is 100M uncompressed.

ref #28435

### PR DESCRIPTION
**Pull request by ixdy**
Fixes #28635. Also simplifies some logic in the kubemark scripts by using a utility function from hack/lib.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33409)

### CODE REVIEW

**Comment by wojtek-t** [comment]
@gmarek - you know this code much better than me, can you please take a look?

File: hack/lib/golang.sh [hunk 1]
```diff
@@ -146,9 +145,11 @@ readonly KUBE_TEST_PORTABLE=(
   hack/lib
 )
 
-# Node test has built-in etcd and kube-apiserver, it can only be built on the
-# same platforms with kube-apiserver.
+# Test targets which run on the Kubernetes clusters directly, so we only
+# need to target server platforms.
+# These binaries will be distributed in the kubernetes-test tarball.
 readonly KUBE_NODE_TEST_TARGETS=(
+  cmd/kubemark
```
Comment (gmarek) [inline hack/lib/golang.sh]: Why node_test_targets? It's only on a machine from which the tests are run, to push the binary to gcr.io

**Comment by gmarek** [comment]
One comment. Generally LGTM.

**Comment by gmarek** [comment]
Recent kubemark failures are of the form:
```
2016/09/26 10:17:17 e2e.go:466: Running: Start Kubemark
Copying kubemark to ./test/kubemark/../../cluster/kubemark/../../cluster/gce/../../cluster/gce/../../cluster/../cluster/images/kubemark
tar (child): kubernetes-server-linux-amd64.tar.gz: Cannot open: No such file or directory
```

@ixdy - do you think this PR can help with that?

File: hack/lib/golang.sh (hunk 1 above)
Comment (ixdy) [inline hack/lib/golang.sh]: Unless I'm really misunderstanding this, `cmd/kubemark` is the hollow-node binary, which gets built into the kubemark container image (cluster/images/kubemark) which is then run on the kubernetes cluster under test, hence we should target platforms that run kubernetes.

I could probably rename this target to `KUBE_SERVER_TEST_TARGETS` if it'd be a little clearer. That'd at least match the other names here (KUBE_SERVER and KUBE_CLIENT).

**Comment by ixdy** [comment]
@gmarek it might. the function in hack/lib is a little bit more robust about finding release artifacts.

**Comment by ixdy** [comment]
`s/KUBE_NODE_TEST/KUBE_TEST_SERVER/`

PTAL

**Comment by ixdy** [comment]
@k8s-bot build this

**Comment by ixdy** [comment]
@mtaufen @fejta out of quota in k8s-jkns-pr-gci-kubemark:
```
ERROR: (gcloud.compute.networks.create) Some requests did not succeed:
 - Quota 'NETWORKS' exceeded. Limit: 5.0

```

**Comment by gmarek** [comment]
GCI Kubemark failure is a suite configuration problem. Merging, as it may help with other kubemark failures.

**Comment by wojtek-t** [comment]
>
Merging, as it may help with other kubemark failures.

@gmarek - can you please clarify this? I don't understand how it can help in general.

**Comment by gmarek** [comment]
@wojtek-t - once in a blue moon Kubemark tests were failing because they weren't able to find a release tar. @ixdy claims that this method of finding them is more robust. See https://github.com/kubernetes/kubernetes/pull/33409#issuecomment-249642321

**Comment by wojtek-t** [comment]
Got it - thanks!

**Comment by ixdy** [comment]
The once-in-a-blue-moon failures are a little weird, though. I'd expect consistent results finding the server tarballs: either we fail repeatedly, since something is misconfigured, or it always works.

**Pull request merged**